compare-bug-version-and-git-branch.sh

```diff
@@ -53,29 +56,31 @@ while [ -z "${BUG_PRODUCT}" -a ${BZQTRY} -le 3 ]; do
     BUG_STATUS=$(cut -d: -f4 <<< "${BZQOUT}")
 done
 
-if [ "${BUG_STATUS}" != "NEW" -a "${BUG_STATUS}" != "POST" -a "${BUG_STATUS}" != "ASSIGNED" ]; then
-    echo "BUG id ${BUG} has an invalid status as ${BUG_STATUS}. Acceptable status values are NEW, ASSIGNED or POST."
+if [ "${BUG_PRODUCT}" != "GlusterFS" ]; then
+    echo "BUG id ${BUG} belongs to '${BUG_PRODUCT}' and not 'GlusterFS'."
     exit 1
 fi
 
-if [ "${BUG_GROUPS}" != '[]' ]; then
-    echo "BUG id ${BUG} is marked private, please remove the groups."
+if [ "${BUG_STATUS}" != "NEW" -a "${BUG_STATUS}" != "POST" -a "${BUG_STATUS}" != "ASSIGNED" -a "${BUG_STATUS}" != "MODIFIED" ]; then
+    echo "BUG id ${BUG} has an invalid status as ${BUG_STATUS}. Acceptable status values are NEW, ASSIGNED, POST or MODIFIED"
     exit 1
 fi
 
-if [ "${BUG_PRODUCT}" != "GlusterFS" ]; then
-    echo "BUG id ${BUG} belongs to '${BUG_PRODUCT}' and not 'GlusterFS'."
+if [ "${BUG_GROUPS}" != '[]' ]; then
+    echo "BUG id ${BUG} is marked private, please remove the groups."
     exit 1
 fi
 
 if [ "${GERRIT_BRANCH}" = "master" ]; then
     if [ "${BUG_VERSION}" != 'mainline' -a \
-         "${BUG_VERSION}" != 'pre-release' ]; then
-            echo "Change filed against the '${GERRIT_BRANCH} branch, but the BUG id is for '${BUG_VERSION}'"
-            exit 1
+                          "${BUG_VERSION}" != 'pre-release' ]; then
```

Comment:
There is no pre-release version field in BZ anymore, not sure when that was removed, but it cannot be BUG_VERSION anymore, so this last case can be deleted IMO.

Reply:
Done